spec-main/extensions-spec.ts

```diff
@@ -175,9 +190,98 @@ describe('chrome extensions', () => {
     });
   });
 
+  describe('chrome.webRequest', () => {
+    async function fetch (contents: WebContents, url: string) {
+      return contents.executeJavaScript(`fetch("${url}")`);
+    }
+
+    describe('onBeforeRequest', () => {
+      it('can cancel http requests', async () => {
+        const customSession = session.fromPartition(`persist:${uuid.v4()}`);
+        const w = new BrowserWindow({ show: false, webPreferences: { session: customSession, sandbox: true } });
+
+        w.webContents.once('dom-ready', async () => {
+          await customSession.loadExtension(path.join(fixtures, 'extensions', 'chrome-webRequest'));
+
+          await expect(fetch(w.webContents, url)).to.eventually.be.rejectedWith(TypeError);
+        });
+
+        w.loadURL(url);
```

Comment:
All `expect`ations should be at the top level, otherwise the test runner will not continue running tests after a failure.

```suggestion
        w.loadURL(url);
        await emittedOnce(w.webContents, 'dom-ready')
        await customSession.loadExtension(path.join(fixtures, 'extensions', 'chrome-webRequest'));
        await expect(fetch(w.webContents, url)).to.eventually.be.rejectedWith(TypeError);
```